### PR TITLE
feat(token validation): clear credentials and redirect to login only …

### LINKED ISSUE #32: Clear credentials automatically...
In the function ```handleTokenValidation``` in case of rejection the service clear the credentials in any case and redirect to login.

What if token service respond with 500? the user is logged out also if "is not his fault".

I will check for specific status code = 401

### PR DESCRIPTION
**Pull request by dami-gg**
…if token is unexistent or invalid (401 error)

Closes #32

### CODE REVIEW

**Comment by gregdeane** [comment]
@dami-gg Because the tabbing was changed (thank you by the way!!) I can't readily tell what changes you made that are specific to this PR. Can you please provide link(s) to the lines in question?

File: src/authentication-service.js [hunk 1]
```diff
@@ -297,50 +297,59 @@ function authenticationFactory($q, $localStorage, $cookies, environmentsService,
    * @returns {*}
    */
   function handleTokenValidation(token, redirectUrl) {
```
Comment (dami-gg) [inline src/authentication-service.js]: @gregdeane I have basically refactored this method, adding the new ones prevalidateToken and clearCredentialsAndRedirectToLogin, plus adapting the tests to cover those

**Comment by dami-gg** [comment]
:+1:

**Comment by gregdeane** [comment]
Pulled branch, ran unit tests. All tests pass, coverage is green.

Per @dami-gg, we are not able to simulate these conditions for manual testing so no manual tests were performed.

**Comment by gregdeane** [comment]
👍

**Pull request merged**